src/typing/Typing.ml

```diff
@@ -53,15 +53,27 @@ let rec type_of_expr lang e : G.name Type.t * G.ident option =
       let t2, _id = type_of_expr lang e2 in
       let t =
         match (t1, op, t2) with
+        | Type.Builtin Type.Int, (G.Plus | G.Minus (* TODO more *)), _
+        | _, (G.Plus | G.Minus (* TODO more *)), Type.Builtin Type.Int
+          when not (Lang.is_js lang) ->
+            (* THINK: Is this correct in every other language that we support? *)
```

Comment:
No, it isn't. At the very least, Java also does implicit coercions to string when an int and a string are operands to `+`. I bet other JVM languages do this too.

Reply:
True, I should have known about Java.

Reply:
Thanks for flagging this!